src/core/shape_inference/include/topk_shape_inference.hpp

```diff
@@ -18,9 +18,6 @@ void shape_infer(const TopK* op,
                  const std::vector<T>& input_shapes,
                  std::vector<T>& output_shapes,
                  const std::map<size_t, std::shared_ptr<ngraph::runtime::HostTensor>>& constant_data = {}) {
-    using DimType = typename std::iterator_traits<typename T::iterator>::value_type;
-    constexpr bool is_dynamic_shape = std::is_base_of<ov::PartialShape, T>::value;
```

Comment:
@usstq  @jane-intel  Please review this change too.

Reply:
This will be covered in a separate code review by @jane-intel